Libraries/Arduino_BNO055/BNO055_IMU.h

```diff
@@ -0,0 +1,172 @@
+#define LIA_DATA_X_MSB      0x29
+#define LIA_DATA_Y_LSB      0x2A
+#define LIA_DATA_Y_MSB      0x2B
+#define LIA_DATA_Z_LSB      0x2C
+#define LIA_DATA_Z_MSB      0x2D
+#define GRV_DATA_X_LSB      0x2E
+#define GRV_DATA_X_MSB      0x2F
+#define GRV_DATA_Y_LSB      0x30
+#define GRV_DATA_Y_MSB      0x31
+#define GRV_DATA_Z_LSB      0x32
+#define GRV_DATA_Z_MSB      0x33
+#define TEMP                0x34
+#define CALIB_STAT          0x35
+#define ST_RESULT           0x36
+#define INT_STATUS          0x37
+#define SYS_CLK_STATUS      0x38
+#define SYS_STATUS          0x39
+#define SYS_ERR             0x3A
+#define UNIT_SEL            0x3B
+#define OPR_MODE            0x3D
+#define PWR_MODE            0x3E
+#define SYS_TRIGGER         0x3F
+#define TEMP_SOURCE         0x40
+#define AXIS_MAP_CONFIG     0x41
+#define AXIS_MAP_SIGN       0x42
+#define ACC_OFFSET_X_LSB    0x55
+#define ACC_OFFSET_X_MSB    0x56
+#define ACC_OFFSET_Y_LSB    0x57
+#define ACC_OFFSET_Y_MSB    0x58
+#define ACC_OFFSET_Z_LSB    0x59
+#define ACC_OFFSET_Z_MSB    0x5A
+#define MAG_OFFSET_X_LSB    0x5B
+#define MAG_OFFSET_X_MSB    0x5C
+#define MAG_OFFSET_Y_LSB    0x5D
+#define MAG_OFFSET_Y_MSB    0x5E
+#define MAG_OFFSET_Z_LSB    0x5F
+#define MAG_OFFSET_Z_MSB    0x60
+#define GYR_OFFSET_X_LSB    0x61
+#define GYR_OFFSET_X_MSB    0x62
+#define GYR_OFFSET_Y_LSB    0x63
+#define GYR_OFFSET_Y_MSB    0x64
+#define GYR_OFFSET_Z_LSB    0x65
+#define GYR_OFFSET_Z_MSB    0x66
+#define ACC_RADIUS_LSB      0x67
+#define ACC_RADIUS_MSB      0x68
+#define MAG_RADIUS_LSB      0x69
+#define MAG_RADIUS_MSB      0x6A
+
+// page 1
+#define PAGE_ID             0x07
+#define ACC_CONFIG          0x08
+#define MAG_CONFIG          0x09
+#define GYRO_CONFIG_0       0x0A
+#define GYRO_CONFIG_1       0x0B
+#define ACC_SLEEP_CONFIG    0x0C
+#define GYR_SLEEP_CONFIG    0x0D
+#define INT_MSK             0x0F
+#define INT_EN              0x10
+#define ACC_AM_THRES        0x11
+#define ACC_INT_SETTINGS    0x12
+#define ACC_HG_DURATION     0x13
+#define ACC_HG_THRESH       0x14
+#define ACC_NM_THRESH       0x15
+#define ACC_NM_SET          0x16
+#define GYR_INT_SETTINGS    0x17
+#define GYR_HR_X_SET        0x18
+#define GYR_DUR_X           0x19
+#define GYR_HR_Y_SET        0x1A
+#define GYR_DUR_Y           0x1B
+#define GYR_HR_Z_SET        0x1C
+#define GYR_DUR_Z           0x1D
+#define GYR_AM_THRESH       0x1E
+#define GYR_AM_SET          0x1F
+
+typedef enum {
+    GYRO, GRAVITY, ACCEL, EUL, QUAT
+} DataMode;
+
+typedef union axes {
+    int16_t accelAxes[3];   // accelerometer sensor output
+    int16_t gyroAxes[3];    // gyro sensor output
+    int16_t gravAxes[3];    // gravity vector output
+    int16_t eulerAngles[3]; // euler angles
+    int16_t quaternions[4]; // qauaternion
+} AxisData;
+
+// a struct of useful data
+typedef struct data {
+    float accelAxes[3];
+    float gyroAxes[3];
+    float gravAxes[3];
+    float eulerAngles[3];
+    float orientation[3];
+} Data;
```

Comment:
Try coming up with a more descriptive name than this. Keep in mind the accelerometer will be included in a board-level file where "Data" could refer to a lot of different sensor readings.

Reply:
Fair enough, I'll make it IMUData